metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -838,6 +840,28 @@ def _make_select_columns_for_multiple_metrics(
                         column_alias=metric_column_name,
                     )
                 )
+        for table_alias, measure_instances in table_alias_to_measure_instances.items():
+            for measure_instance in measure_instances:
+                measure_spec = measure_instance.spec
+                measure_column_name = self._column_association_resolver.resolve_spec(measure_spec).column_name
+                column_reference_expression = SqlColumnReferenceExpression(
+                    col_ref=SqlColumnReference(
+                        table_alias=table_alias,
+                        column_name=measure_column_name,
+                    )
+                )
+                if aggregation_type:
+                    select_expression: SqlExpressionNode = SqlFunctionExpression.build_expression_from_aggregation_type(
+                        aggregation_type=aggregation_type, sql_column_expression=column_reference_expression
+                    )
+                else:
+                    select_expression = column_reference_expression
+                select_columns.append(
+                    SqlSelectColumn(
+                        expr=select_expression,
+                        column_alias=measure_column_name,
+                    )
+                )
```

Comment:
Ok, I see. Since this is in fact identical to the metric logic I think we can do a little refactor to have a MetricInput Protocol type (or whatever we want to call it) and work off of that.

Do we need to apply the fill nulls logic here as well? I suspect we do, since this is a post-aggregation join, right? Assuming that's the case, we can add an is_measure property to the protocol if we need to.

Reply:
> Do we need to apply the fill nulls logic here as well?

This is a good point actually. Need to figure out how we can propagate that information for combining AggregateMeasureNodes as it's easy for ComputeMetricNodes when you have the metric_spec and you can get the measure_inputs from that, but in this case we're just combining the AggregateMeasureNodes which means we need to somehow pass the metric_spec without actually using it during the combine process

Reply:
Oh wait, the `AggregateMeasureNodes` has `metric_input_measure_specs` in it.